riley-research/IPSA
# Adjust calculations of precursorMass and precursorMz

### Summary of changes

- Refactor `MainActivity` component to use `PeptideStore` for fetching plot data instead of static test data.
- Modify the helper function `generateFragments` to reverse the loop condition in charge state iteration.
- Consolidate and simplify fragment mass calculation logic by creating a helper function `calculateDeltaMassForType`.
- Added logging in `CalculatePrecursorMass` in PHP for debugging.

### How to test if necessary

1. Run the application and ensure the `MainActivity` component displays plot data accurately based on the selected peptide from `PeptideStore`.
2. Check the fragment generation to ensure it includes intended fragment types and their order.
3. Verify no errors are thrown during `CalculatePrecursorMass` logging in PHP.

These changes were made to improve the maintainability and readability of the fragment generation process by simplifying and consolidating the mass calculation logic and using MobX for state management in the `MainActivity` component.
